src/mlpack/methods/ann/loss_functions/margin_ranking_loss_impl.hpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * @file margin_ranking_loss_impl.hpp
+ * @author Andrei Mihalea
+ *
+ * Implementation of the Margin Ranking Loss function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTION_MARGIN_IMPL_LOSS_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTION_MARGIN_IMPL_LOSS_HPP
+
+// In case it hasn't been included.
+#include "margin_ranking_loss.hpp"
+
+namespace mlpack {
+namespace ann /** Artifical Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+MarginRankingLoss<InputDataType, OutputDataType>::MarginRankingLoss(
+    const double margin) : margin(margin)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template <
+    typename InputType,
+    typename TargetType
+>
+double MarginRankingLoss<InputDataType, OutputDataType>::Forward(
+    const InputType& input,
+    const TargetType& target)
+{
+  int input_rows = input.n_rows;
```

Comment:
As per the style guidelines, we use camel casing for all out variables
```suggestion
  const int inputRows = input.n_rows;
```